From b2b2002a5975595819ef04c2b941c4c3962c6bcd Mon Sep 17 00:00:00 2001
Message-Id: <b2b2002a5975595819ef04c2b941c4c3962c6bcd.1350309834.git.minovotn@redhat.com>
In-Reply-To: <14b740b7e3fbafdbfe2e6a302f51d6aa4fa96bf7.1350309834.git.minovotn@redhat.com>
References: <14b740b7e3fbafdbfe2e6a302f51d6aa4fa96bf7.1350309834.git.minovotn@redhat.com>
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Wed, 10 Oct 2012 17:52:31 +0200
Subject: [PATCH 04/17] add missing CPUID bit constants

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1349891552-23115-2-git-send-email-ehabkost@redhat.com>
Patchwork-id: 42935
O-Subject: [RHEL6.4 qemu-kvm PATCH 1/2] add missing CPUID bit constants
Bugzilla: 843084
RH-Acked-by: Marcelo Tosatti <mtosatti@redhat.com>
RH-Acked-by: Gleb Natapov <gleb@redhat.com>
RH-Acked-by: Avi Kivity <avi@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>

Bugzilla: 843084
Upstream status: pending
 (this will be sent only when we send the Haswell model definition upstream)

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 target-i386/cpu.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 target-i386/cpu.h | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/target-i386/cpu.h b/target-i386/cpu.h
index 7dfcbda..a0673d5 100644
--- a/target-i386/cpu.h
+++ b/target-i386/cpu.h
@@ -388,6 +388,7 @@
 #define CPUID_EXT_TM2      (1 << 8)
 #define CPUID_EXT_SSSE3    (1 << 9)
 #define CPUID_EXT_CID      (1 << 10)
+#define CPUID_EXT_FMA      (1 << 12)
 #define CPUID_EXT_CX16     (1 << 13)
 #define CPUID_EXT_XTPR     (1 << 14)
 #define CPUID_EXT_PDCM     (1 << 15)
@@ -447,7 +448,18 @@
 #define CPUID_EXT3_SKINIT  (1 << 12)
 #define CPUID_EXT3_FMA4    (1 << 16)
 
+
+#define CPUID_7_0_EBX_FSGSBASE (1 << 0)
+#define CPUID_7_0_EBX_BMI1     (1 << 3)
+#define CPUID_7_0_EBX_HLE      (1 << 4)
+#define CPUID_7_0_EBX_AVX2     (1 << 5)
 #define CPUID_7_0_EBX_SMEP     (1 << 7)
+#define CPUID_7_0_EBX_BMI2     (1 << 8)
+#define CPUID_7_0_EBX_ERMS     (1 << 9)
+#define CPUID_7_0_EBX_INVPCID  (1 << 10)
+#define CPUID_7_0_EBX_RTM      (1 << 11)
+#define CPUID_7_0_EBX_RDSEED   (1 << 18)
+#define CPUID_7_0_EBX_ADX      (1 << 19)
 #define CPUID_7_0_EBX_SMAP     (1 << 20)
 
 #define CPUID_VENDOR_INTEL_1 0x756e6547 /* "Genu" */
-- 
1.7.11.7

